javascript/javascript2/week2/Working with movies.js

```diff
@@ -0,0 +1,94 @@
+{"title": "101 Dalmatians","year": 1996,"rating": 5.7,"votes": 83245,"running_times": 6180},
+{"title": "102 Dalmatians","year": 2000,"rating": 4.9,"votes": 28927,"running_times": 6000},
+{"title": "11-11-11","year": 2011,"rating": 4,"votes": 11425,"running_times": 5400},
+{"title": "11:14","year": 2003,"rating": 7.2,"votes": 40149,"running_times": 5700},
+{"title": "12 Angry Men","year": 1957,"rating": 8.9,"votes": 518449,"running_times": 5760},
+{"title": "12 Rounds","year": 2009,"rating": 5.6,"votes": 24457,"running_times": 6480}
+]
+
+//Create an array of movies containing the movies with a short title (you define what short means)
+const shortestTitel = movies.filter(movie => movie.title.length <= 2)
+
+console.log(shortestTitel)
+
+// Create an array of movie titles with long movie titles
+
+const longMovieTitle = movies.filter(movie => movie.title.length > 21)
+
+console.log(longMovieTitle)
+
+
+//Count the number of movies made between 1980-1989 (including both the years)
+
+
+let CountMovies1980_1989 = movies.filter(movies => movies.year >= 1980 && movies.year <= 1989)
+console.log(CountMovies1980_1989)
+
+
+
+
+//Create a new array that has an extra key called tag. The tag is based on the rasist: Good (>= 7), Average (>= 4 and < 7), Bad (< 4)
+
+//console.log(movies.map(key => {key.tag = 'tag'; return key;}))
+
+const Tag = movies.map((movie) => {
+ let rating =
+  movie.rating >= 7
+     ? "Good"
+     : movie.rating >= 4 && movie.rating < 7
+     ? "Average"
+     :"Bad" 
+movie.tag = rating;
+ return movie;
+});
+console.log(Tag);
+
+
+//Using chaining, first filter the movies array to only contain the movies rated higher than 6. Now map the movies array to only the rating of the movies.const highRatingMovies = movies
+const highRateMovies = movies.filter(movies => movies.rating > 6)
+
+console.log(highRateMovies)
+//Count the total number of movies containing any of following keywords: Surfer, Alien or Benjamin. So if there were 3 movies that contained Surfer, 1 with Alien and 2 with Benjamin, you would return 6. Can you make sure the search is case insensitive?
+
+
+const moviesWithKeyword = movies.filter(movie => 
+    movie.title.match(/Surfer/i)).length +
+    movies.filter(movie => movie.title.match(/Benjamin/i)).length +
+    movies.filter(movie => movie.title.match(/Alien/i)).length;
+
+console.log(`${moviesWithKeyword} movies contains following keywords: Surfer, Alien or Benjamin.`);
+
+
+//Calculate the average rating of all the movies using reduce
+// let totalRating = 0
+// movies.forEach(movie => {totalRating += movie.rating / movies.length})
+
+// console.log(totalRating);
+
+let totalRating = movies.reduce((total, movie) => {
+ return total + movie.rating / movies.length} , 0)
+
+console.log(totalRating);
+
+//Count the total number of Good, Average and Bad movies using reduce. A return could fx be {goodMovies: 33, averageMovies: 45, goodMovies: 123}
+
```

Comment:
I'll just show you a possible solution!

```suggestion
const reducer = (acc, movie) => {
  const rating = getMovieRating(movie.rating).toLowerCase(); // like above, translate 7 to Good, etc.
  
  const count= acc[rating];
  
  return {
    ...acc,
    [rating]: count + 1,
  };
};
const movieCountByRating = movies.reduce(reducer, {
  good: 0,
  average: 0,
  bad: 0
});
```